packages/graphql-hooks/index.d.ts

```diff
@@ -0,0 +1,118 @@
+// Exports
+
+export class GraphQLClient {
+  constructor(options: ClientOptions)
+
+  cache: Cache
+  headers: Headers
+  ssrMode: boolean
+  fetchOptions: object
+  logErrors: boolean
+
+  private onError(): any
+  private fetch(): Promise<any>
+
+  setHeader(key: string, value: string): GraphQLClient
+  setHeaders(object): GraphQLClient
+  logErrorResult({ result: object, operation: Operation }): void
+  request(operation: Operation, options: object): Promise<any>
+}
+
+export function useClientRequest(
+  query: string,
+  initialOptions?: UseClientRequestOptions
+): [FetchData, UseClientRequestResult]
+
+export function useQuery(
+  query: string,
+  options?: UseQueryOptions
+): UseQueryResult
+
+export function useManualQuery(
+  query: string,
+  options?: UseClientRequestOptions
+): [FetchData, UseClientRequestResult]
+
+export function useMutation(
+  query: string,
+  options?: UseClientRequestOptions
+): [FetchData, UseClientRequestResult]
+
+export const ClientContext: any
+
+// internal types
+
+interface ClientOptions {
+  url: string
+  cache?: Cache
+  headers?: Headers
+  ssrMode?: boolean
+  fetch?(url: string, options?: object): Promise<object>
+  fetchOptions?: object
+  logErrors?: boolean
+  onError?({ operation: Operation, result: Result }): void
+}
+
+type Headers = { [k: string]: string }
+
+interface Cache {
+  get(keyObject: object): object
```

Comment:
Minor: if we define `getCacheKey(operation: Operation, options: UseClientRequestResult)` and it's return type we could use it here.

Reply:
I've added this in the latest commit